common/notifications.cpp

```diff
@@ -787,24 +790,47 @@ namespace rs2
         _failed = true;
     }
 
+    void notification_model::invoke(std::function<void()> action)
+    {
+        single_consumer_queue<bool> q;
+        dispatch_queue.enqueue([&q, &action](){
+            try
+            {
+                action();
+                q.enqueue(true);
+            }
+            catch(...)
+            {
+                q.enqueue(false);
+            }
+        });
+        bool res;
+        if (!q.dequeue(&res, 100000) || !res)
```

Comment:
IIRC firmware backup in USB2 mode takes more than a minute, so it can potentially reach 100 sec as well.

Reply:
Firmware backup would not be executed under invoke (on the main thread).
This bound captures max time between two render frames (UI main loop).
And if it fails on timeout, clean error will be thrown (no undefined behaviour as far as I can tell)